src/components/AttachmentPicker/index.native.js

```diff
@@ -177,30 +178,43 @@ class AttachmentPicker extends Component {
       * Common image picker handling
       *
       * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+      * @param {Boolean} isCamera
       * @returns {Promise<ImagePickerResponse>}
       */
-    showImagePicker(imagePickerFunc) {
+    showImagePicker(imagePickerFunc, isCamera = false) {
         return new Promise((resolve, reject) => {
-            imagePickerFunc(getImagePickerOptions(this.props.type), (response) => {
-                if (response.didCancel) {
-                    // When the user cancelled resolve with no attachment
-                    return resolve();
-                }
-                if (response.errorCode) {
-                    switch (response.errorCode) {
-                        case 'permission':
-                            this.showPermissionsAlert();
-                            break;
-                        default:
-                            this.showGeneralAlert();
-                            break;
+            if (!isCamera) {
+                this.openImagePicker(imagePickerFunc, resolve, reject);
```

Comment:
As mentioned in the last PR too, lets not pass `resolve` and `reject` as params. Instead make it return the promise.